praetorian_cli/sdk/chariot.py

```
@@ -213,9 +213,21 @@ def download(self, name: str, global_=False) -> bytes:
             params |= GLOBAL_FLAG
 
-        resp = self._make_request('GET', self.url('/file'), params=params, allow_redirects=True)
+        # Step 1: Get presigned URL from backend
+        resp = self._make_request('GET', self.url('/file'), params=params)
         process_failure(resp)
-        return resp.content
+
+        # Parse JSON response to get presigned URL
+        presigned_url = resp.json().get('url')
+        if not presigned_url:
+            raise Exception('No download URL returned from server')
```

Ruff (0.14.1)

Create your own exception

(TRY002)